src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -70,22 +72,18 @@ public void updateOrder(String orderId, CompositePurchaseOrder compPO) {
     getPurchaseOrderById(orderId, lang, httpClient, ctx, okapiHeaders, logger)
       .thenCompose(poFromStorage -> validatePoNumber(compPO, poFromStorage))
       .thenCompose(poFromStorage -> updatePoLines(poFromStorage, compPO).thenApply(v -> poFromStorage))
-      .thenAccept(poFromStorage -> {
-        CompletableFuture<Void> updatedPoFuture;
+      .thenCompose(poFromStorage -> {
         if (isTransitionToOpen(compPO, poFromStorage)) {
-          updatedPoFuture = openOrder(compPO);
+          return openOrder(compPO).thenCompose(v -> updateCompositePoLines(compPO));
```

Comment:
You might move `updateCompositePoLines` call to `openOrder` method

Reply:
done